### PR TITLE
nu-table/ Add `-t/theme` argument && Replace `-n/start-number` with `-i/index`

### PR DESCRIPTION
**Pull request by zhiburt**
ref #11054

cc: @fdncred 

I've not figured out how to be able to have a flag option as `table -i` :(

```nu
~/bin/nushell> [[a b, c]; [1 [2 3 3] 3] [4 5 [1 2 [1 2 3]]]] | table -e --width=80 --theme basic -i false

+---+-------+-----------+
| a |   b   |     c     |
+---+-------+-----------+
| 1 | +---+ |         3 |
|   | | 2 | |           |
|   | +---+ |           |
|   | | 3 | |           |
|   | +---+ |           |
|   | | 3 | |           |
|   | +---+ |           |
+---+-------+-----------+
| 4 |     5 | +-------+ |
|   |       | |     1 | |
|   |       | +-------+ |
|   |       | |     2 | |
|   |       | +-------+ |
|   |       | | +---+ | |
|   |       | | | 1 | | |
|   |       | | +---+ | |
|   |       | | | 2 | | |
|   |       | | +---+ | |
|   |       | | | 3 | | |
|   |       | | +---+ | |
|   |       | +-------+ |
+---+-------+-----------+
```

```nu
~/bin/nushell> [[a b, c]; [1 [2 3 3] 3] [4 5 [1 2 [1 2 3]]]] | table -e --width=80 --theme basic -i 100

+-----+---+-------------+-----------------------+
|   # | a |      b      |           c           |
+-----+---+-------------+-----------------------+
| 100 | 1 | +-----+---+ |                     3 |
|     |   | | 100 | 2 | |                       |
|     |   | +-----+---+ |                       |
|     |   | | 101 | 3 | |                       |
|     |   | +-----+---+ |                       |
|     |   | | 102 | 3 | |                       |
|     |   | +-----+---+ |                       |
+-----+---+-------------+-----------------------+
| 101 | 4 |           5 | +-----+-------------+ |
|     |   |             | | 100 |           1 | |
|     |   |             | +-----+-------------+ |
|     |   |             | | 101 |           2 | |
|     |   |             | +-----+-------------+ |
|     |   |             | | 102 | +-----+---+ | |
|     |   |             | |     | | 100 | 1 | | |
|     |   |             | |     | +-----+---+ | |
|     |   |             | |     | | 101 | 2 | | |
|     |   |             | |     | +-----+---+ | |
|     |   |             | |     | | 102 | 3 | | |
|     |   |             | |     | +-----+---+ | |
|     |   |             | +-----+-------------+ |
+-----+---+-------------+-----------------------+
```

### CODE REVIEW

**Comment by zhiburt** [comment]
Theoretically if someone would came up with a good syntax we could set a custom style from string too.
Something like `table --theme="****####XX"` (but it does not recognize a absence of a line)

**Comment by zhiburt** [comment]
Self-note: Yessss there's many places we clone configs....around... which certainly deserve a refactoring.........

**Comment by zhiburt** [comment]
> Theoretically if someone would came up with a good syntax we could set a custom style from string too. Something like `table --theme="****####XX"` (but it does not recognize a absence of a line) But it'd be a niche thing so....

We could use a object by the way, something like

```
| table --theme { top: '_', bottom: '_', left: {top: '+', bottom: '+'} }
```

You got the idea.

**Comment by zhiburt** [comment]
By mistake closed it... sorry

File: crates/nu-command/src/viewers/table.rs [hunk 1]
```diff
@@ -60,12 +65,17 @@ impl Command for Table {
...
+                "set a table mode/theme",
+                Some('t'),
+            )
+            .named(
+                "index",
+                SyntaxShape::Any,
+                "set or set off a index mode/theme",
+                Some('i'),
             )
-            .switch("list", "list available table modes/themes", Some('l'))
```
Comment (fdncred) [inline crates/nu-command/src/viewers/table.rs:68]: I'm not sure why --list is removed here? We still need it to show people what all the different modes/themes are.

File: crates/nu-command/src/viewers/table.rs [hunk 2]
```diff
@@ -111,16 +121,9 @@ impl Command for Table {
         call: &Call,
         input: PipelineData,
     ) -> Result<PipelineData, ShellError> {
-        let list_themes: bool = call.has_flag("list");
```
Comment (fdncred) [inline crates/nu-command/src/viewers/table.rs:114]: Same here

File: crates/nu-command/src/viewers/table.rs [hunk 3]
```diff
@@ -111,16 +121,9 @@ impl Command for Table {
...
         let cfg = parse_table_config(call, engine_state, stack)?;
         let input = CmdInput::new(engine_state, stack, call, input);
 
-        // if list argument is present we just need to return a list of supported table themes
-        if list_themes {
-            let val = Value::list(supported_table_modes(), Span::test_data());
-            return Ok(val.into_pipeline_data());
-        }
-
```
Comment (fdncred) [inline crates/nu-command/src/viewers/table.rs:123]: same here

File: crates/nu-command/src/viewers/table.rs [hunk 4]
```diff
@@ -909,25 +1007,3 @@ where
 
     out
 }
-
-fn supported_table_modes() -> Vec<Value> {
```
Comment (fdncred) [inline crates/nu-command/src/viewers/table.rs:913]: same here

File: crates/nu-command/src/viewers/table.rs (hunk 1 above)
Comment (zhiburt) [reply crates/nu-command/src/viewers/table.rs:68]: It'll be shown if a unknown theme is picked

Comment (zhiburt) [reply crates/nu-command/src/viewers/table.rs:68]:
```nu
~/bin/nushell> ls | table --theme basic222342
Error: nu::shell::cant_convert

  × Can't convert to theme.
   ╭─[source:1:1]
 1 │ nu
   · ▲
   · ╰── can't convert string to theme
   ╰────
  help: expected either 'basic', 'thin', 'light', 'compact',
        'with_love', 'compact_double', 'rounded', 'reinforced',
        'heavy', 'none', 'psql', 'markdown', 'dots', 'restructured',
        'ascii_rounded', or 'basic_compact'
```

Comment (fdncred) [reply crates/nu-command/src/viewers/table.rs:68]: I think it's good to have an error like this when there's a typo, but if you're looking for the complete list of what table modes nushell supports, you should be able to ask for it. If you have to create an error, just to see the list, it's not very discoverable or easy to figure out.

Comment (zhiburt) [reply crates/nu-command/src/viewers/table.rs:68]: restored

Comment (fdncred) [reply crates/nu-command/src/viewers/table.rs:68]: Thank you!

File: crates/nu-command/src/viewers/table.rs [hunk 5]
```diff
@@ -179,30 +190,53 @@ impl Command for Table {
...
                 ])),
             },
+            Example {
+                description: "Change a theme",
+                example: r#"[[a b]; [1 2] [2 [4 4]]] | table --theme basic"#,
+                result: None,
+            },
+            Example {
+                description: "Force a show of an index",
+                example: r#"[[a b]; [1 2] [2 [4 4]]] | table -i"#,
```
[the remark this answers is not in the document]
Comment (fdncred) [reply crates/nu-command/src/viewers/table.rs:200]: I added this for now. https://github.com/nushell/nushell/pull/11071

[the remark this answers is not in the document]
Comment (zhiburt) [reply crates/nu-command/src/viewers/table.rs:200]: You mean `-i` without an argument?

[the remark this answers is not in the document]
Comment (fdncred) [reply crates/nu-command/src/viewers/table.rs:200]: ya, without the argument. So I changed the example to `-i true` or `-i false`.

**Pull request merged**